preprocessing-standards.Rmd

```diff
@@ -134,22 +138,23 @@ cwp <- CentWaveParam(ppm = 50,
 
 data <- findChromPeaks(data, param = cwp) 
 pks <- chromPeaks(data)
-```    
-    
-Consider Xanthine
-```{r}
-mz_xan <- 152.03384
-dev <- 0.01
-xan_pos <- data %>%
-    filterRt(rt = c(130, 150)) %>%
-    filterMz(mz = c(mz_xan + 1 - dev, mz_xan + 1 + dev)) %>%
-    filterFile(file = which(data$polarity == "POS"))
+```
+
+We next evaluate the signal for Xanthine. For this we first calculate first its
+monoisotopic mass based on the compound's chemical formula and then the m/z of
+its [M+H]+ ion. 
+
+```{r xanthine-mz}
+frml_xan <- std_dilution01[std_dilution01$name == "Xanthine", "formula"]
+mass_xan <- getMolecule(frml_xan)$exactmass    # calculate mass
+mz_xan <- mass2mz(mass_xan, "[M+H]+")[1, 1]    # get the m/z for [M+H]+
+mzr_xan <- mz_xan + c(1, -1) * ppm(mz_xan, 50) # calculate m/z range
 ```
 
 
 ```{r Xanthine-detected_peaks, fig.path = IMAGE_PATH, fig.cap = "Detected peaks for Xanthine.", fig.width = 7, fig.height = 5}
 chr <- chromatogram(filterFile(data, file = which(data$polarity == "POS")), 
-                    rt = c(130,150), mz = c(mz_xan + 1 - dev, mz_xan + 1 + dev))
+                    rt = c(130, 150), mz = mzr_xan)
```

Comment:
instead of a +/- 0.01 difference we usually use a *ppm* (parts-per-million) difference. the ppm is dependent on the actual m/z and that better resembles how our MS instruments works. It has a measurement error on the m/z which is dependent on the m/z: higher m/z values will have a higher error.